exist-core/src/main/java/org/exist/xquery/value/Type.java

```diff
@@ -41,83 +42,105 @@
-    public final static int STRING = 22;
-    public final static int BOOLEAN = 23;
-    public final static int QNAME = 24;
-    public final static int ANY_URI = 25;
-    public final static int BASE64_BINARY = 26;
-    public final static int HEX_BINARY = 27;
-    public final static int NOTATION = 28;
-    public final static int NUMBER = 30;
-    public final static int INTEGER = 31;
-    public final static int DECIMAL = 32;
-    public final static int FLOAT = 33;
-    public final static int DOUBLE = 34;
-    public final static int NON_POSITIVE_INTEGER = 35;
-    public final static int NEGATIVE_INTEGER = 36;
-    public final static int LONG = 37;
-    public final static int INT = 38;
-    public final static int SHORT = 39;
-    public final static int BYTE = 40;
-    public final static int NON_NEGATIVE_INTEGER = 41;
-    public final static int UNSIGNED_LONG = 42;
-    public final static int UNSIGNED_INT = 43;
-    public final static int UNSIGNED_SHORT = 44;
-    public final static int UNSIGNED_BYTE = 45;
-    public final static int POSITIVE_INTEGER = 46;
-    public final static int DATE_TIME = 50;
-    public final static int DATE = 51;
-    public final static int TIME = 52;
-    public final static int DURATION = 53;
-    public final static int YEAR_MONTH_DURATION = 54;
-    public final static int DAY_TIME_DURATION = 55;
-    public final static int GYEAR = 56;
-    public final static int GMONTH = 57;
-    public final static int GDAY = 58;
-    public final static int GYEARMONTH = 59;
-    public final static int GMONTHDAY = 71;
-    public final static int DATE_TIME_STAMP = 72;
-    public final static int TOKEN = 60;
-    public final static int NORMALIZED_STRING = 61;
-    public final static int LANGUAGE = 62;
-    public final static int NMTOKEN = 63;
-    public final static int NAME = 64;
-    public final static int NCNAME = 65;
-    public final static int ID = 66;
-    public final static int IDREF = 67;
-    public final static int ENTITY = 68;
-    public final static int JAVA_OBJECT = 100;
-    public final static int FUNCTION_REFERENCE = 101;
-    public final static int MAP = 102;
-    public final static int ARRAY = 103;
     private final static Logger LOG = LogManager.getLogger(Type.class);
 
-    private static int NO_SUCH_VALUE = -99;
-
-    private final static int[] superTypes = new int[512];
-    private final static Int2ObjectOpenHashMap<String[]> typeNames = new Int2ObjectOpenHashMap<>(64, Hash.FAST_LOAD_FACTOR);
-    private final static Object2IntOpenHashMap<String> typeCodes = new Object2IntOpenHashMap<>(100, Hash.FAST_LOAD_FACTOR);
+    // Tombstone value used for FastUtil Int maps
+    private static final int NO_SUCH_VALUE = -99;
+
+    public final static int ITEM = 1;
+
+    public final static int ANY_TYPE = 2;
+
+    public final static int ANY_SIMPLE_TYPE = 3;
+
+    /* xs:anyAtomicType and its subtypes */
+    public final static int ANY_ATOMIC_TYPE = 4;
```

Comment:
> Just a question: why this rename from ATOMIC ? Is this more in line with the (XDM?) spec?

@dizzzz Yes, exactly - https://www.w3.org/TR/xpath-datamodel-31/#anyAtomicType

> thinking a nit more (from the top of my head) that this might break a lot of existing (java) extension (if they are already not broken :-P )

@dizzzz I doubt many of them are using `ATOMIC` as it is a very generic type. Also this is targeting 7.x.x where we are allowed to improve things that might break things.

Reply:
my mistake, sorry, I thought that more names were changed.